app/src/main/java/com/amaze/filemanager/ui/fragments/preferencefragments/BackupPrefsFragment.kt

```diff
@@ -0,0 +1,198 @@
+import com.amaze.filemanager.R
+import com.amaze.filemanager.TagsHelper
+import com.amaze.filemanager.ui.activities.MainActivity
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import java.io.*
+
+class BackupPrefsFragment : BasePrefsFragment() {
+
+    private val TAG: String = TagsHelper.getTag(BasePrefsFragment::class.java)
+    private val IMPORT_BACKUP_FILE: Int = 2
+
+    override val title = R.string.backup
+
+    private val onExportPrefClick = OnPreferenceClickListener {
+
+        val map: Map<String?, *> = PreferenceManager.getDefaultSharedPreferences(getActivity()).all
+
+        val gsonString: String = Gson().toJson(map)
+
+        try {
+
+            val file = File(context?.cacheDir?.absolutePath + File.separator + "amaze_backup.json")
+
+            val fileWriter = FileWriter(file)
+
+            fileWriter.append(gsonString)
+
+            Log.e(TAG, "wrote export to: ${file.absolutePath}")
+
+            fileWriter.flush()
+            fileWriter.close()
+
+            Toast.makeText(
+                context,
+                getString(R.string.select_save_location),
+                Toast.LENGTH_SHORT
+            ).show()
+
+            val intent = Intent(context, MainActivity::class.java)
+
+            intent.action = Intent.ACTION_SEND
+            intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
+
+            intent.putExtra(Intent.EXTRA_STREAM, Uri.fromFile(file))
+
+            startActivity(intent)
+        } catch (e: IOException) {
+            Toast.makeText(context, getString(R.string.exporting_failed), Toast.LENGTH_SHORT).show()
+            e.printStackTrace()
+        }
+
+        true
+    }
+
+    private val onImportPrefClick = OnPreferenceClickListener {
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+            startActivityForResult(
+                Intent(Intent.ACTION_OPEN_DOCUMENT)
+                    .addCategory(Intent.CATEGORY_OPENABLE)
+                    .setType("*/*")
+                    .putExtra(
+                        Intent.EXTRA_MIME_TYPES,
+                        arrayOf("application/json")
+                    ),
+                IMPORT_BACKUP_FILE
+            )
+        } else {
+            startActivityForResult(
+                Intent.createChooser(
+                    Intent(Intent.ACTION_GET_CONTENT)
+                        .setType("application/json"),
+                    "Choose backup file"
+                ),
+                IMPORT_BACKUP_FILE
+            )
+        }
+
+        true
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+
+        if (requestCode == IMPORT_BACKUP_FILE &&
+            resultCode == Activity.RESULT_OK &&
+            data != null && data.data != null
+        ) {
+            val uri = data.data
+
+            Log.e(TAG, "read import file: $uri")
+
+            try {
```

Comment:
This has a few try catch nested, you should collect all errors and provide the user with error based feedback. Maybe print the exception message to a toast?